pkg/server/tests/commontest/tidb_test.go

```diff
@@ -3692,23 +3692,28 @@ func TestAuditPluginRetrying(t *testing.T) {
 		_, err = db.Exec("INSERT INTO auto_retry_test VALUES (1, 0)")
 		require.NoError(t, err)
 
-		testResults = testResults[:0]
 		// a big enough concurrency to trigger retries
 		concurrency := 500
-		var wg sync.WaitGroup
-		for range concurrency {
-			wg.Add(1)
-			conn, err := db.Conn(context.Background())
-			require.NoError(t, err)
-			go func() {
-				defer wg.Done()
-				_, err := conn.QueryContext(context.Background(), "UPDATE auto_retry_test SET val = val + 1 WHERE id = 1")
+		// Usually the following retry-loop will succeed in the first try. However, if we are lucky
+		// enough, it might need more times to trigger the retry.
+		require.Eventually(t, func() bool {
+			testResults = testResults[:0]
+			var wg sync.WaitGroup
+			for range concurrency {
+				wg.Add(1)
+				conn, err := db.Conn(context.Background())
 				require.NoError(t, err)
-			}()
-		}
-		wg.Wait()
+				go func() {
+					defer wg.Done()
```

Comment:
note that we can use https://pkg.go.dev/sync#WaitGroup.Go now. Don't need to write wg.Add, go func(), defer wg.Done()